servlet/src/main/java/com/linecorp/armeria/server/servlet/DefaultServletContext.java

```diff
@@ -0,0 +1,570 @@
+    public Enumeration<Servlet> getServlets() {
+        final List<Servlet> list = new ArrayList<>();
+        for (ServletRegistration registration : servletRegistrationMap.values()) {
+            list.add(registration.getServlet());
+        }
+        return Collections.enumeration(list);
+    }
+
+    @Override
+    public Enumeration<String> getServletNames() {
+        final List<String> list = new ArrayList<>();
+        for (ServletRegistration registration : servletRegistrationMap.values()) {
+            list.add(registration.getName());
+        }
+        return Collections.enumeration(list);
+    }
+
+    @Override
+    public void log(String msg) {
+        requireNonNull(msg, "msg");
+        logger.debug(msg);
+    }
+
+    @Override
+    public void log(Exception exception, String msg) {
+        requireNonNull(exception, "exception");
+        requireNonNull(msg, "msg");
+        logger.debug(msg,exception);
+    }
+
+    @Override
+    public void log(String message, Throwable throwable) {
+        requireNonNull(message, "message");
+        requireNonNull(throwable, "throwable");
+        logger.debug(message,throwable);
+    }
+
+    @Override
+    public String getServerInfo() {
+        return ServletUtil.getServerInfo()
+                .concat("(JDK ")
+                .concat(ServletUtil.getJvmVersion())
+                .concat(";")
+                .concat(ServletUtil.getOsName())
+                .concat(" ")
+                .concat(ServletUtil.getArch())
+                .concat(")");
+    }
+
+    @Override
+    public String getInitParameter(String name) {
+        requireNonNull(name, "name");
+        return initParamMap.get(name);
+    }
+
+    @Override
+    public Enumeration<String> getInitParameterNames() {
+        return Collections.enumeration(initParamMap.keySet());
+    }
+
+    @Override
+    public boolean setInitParameter(String name, String value) {
+        requireNonNull(name, "name");
+        requireNonNull(value, "value");
+        return initParamMap.putIfAbsent(name,value) == null;
+    }
+
+    @Override
+    public Object getAttribute(String name) {
+        requireNonNull(name, "name");
+        return attributeMap.get(name);
+    }
+
+    @Override
+    public Enumeration<String> getAttributeNames() {
+        return Collections.enumeration(attributeMap.keySet());
+    }
+
+    @Override
+    public void setAttribute(String name, @Nullable Object object) {
+        requireNonNull(name);
+        if (object == null) {
+            removeAttribute(name);
+            return;
+        }
+        attributeMap.put(name,object);
+    }
+
+    @Override
+    public void removeAttribute(String name) {
+        requireNonNull(name, "name");
+        attributeMap.remove(name);
+    }
+
+    @Override
+    public String getServletContextName() {
+        return servletContextName;
+    }
+
+    @Override
+    public ServletRegistration addServlet(String servletName, String className) {
```

Comment:
Question: Should we allow to add a servlet after the `Server` is started?

Reply:
Probably not until there's a demand for such feature? It'd be nice to keep the supported feature set to an essential extent.

Reply:
Agree, I will add started flag to reject servlet adding after the Server is started.